packages/client/browser/index.ts

```diff
@@ -46,34 +47,41 @@ export * from './logging'
 import { getLogger } from './logging'
 
 export async function createClient(args: any) {
-  const logger = getLogger({ loglevel: args.loglevel ?? 'info' })
+  const logger = getLogger({ loglevel: args.loglevel })
   const datadir = args.datadir ?? Config.DATADIR_DEFAULT
   const common = new Common({ chain: args.network ?? Chain.Mainnet })
   const key = await Config.getClientKey(datadir, common)
+  const bootnodes = args.bootnodes ? parseMultiaddrs(args.bootnodes) : undefined
   const config = new Config({
     common,
     key,
-    servers: [new Libp2pServer({ multiaddrs: [], config: new Config({ key, logger }), ...args })],
-    syncmode: args.syncmode ?? 'full',
+    transports: ['libp2p'],
+    syncmode: args.syncmode,
+    bootnodes,
+    multiaddrs: [],
     logger,
+    maxPerRequest: args.maxPerRequest,
+    minPeers: args.minPeers,
+    maxPeers: args.maxPeers,
+    discDns: false,
   })
-  return new EthereumClient({ config })
+  config.events.setMaxListeners(50)
+  const chainDB = level(common.chainName())
+  return new EthereumClient({ config, chainDB })
 }
 
 export async function run(args: any) {
   const client = await createClient(args)
-  const { logger, chainCommon: common } = client.config
-  logger.info('Initializing Ethereumjs client...')
-  logger.info(`Connecting to network: ${common.chainName()}`)
+  const { logger, chainCommon } = client.config
+  logger.info(`Initializing Ethereumjs client network=${chainCommon.chainName()}`)
   client.config.events.on(Event.SERVER_ERROR, (err) => logger.error(err))
   client.config.events.on(Event.SERVER_LISTENING, (details) => {
     logger.info(`Listener up transport=${details.transport} url=${details.url}`)
   })
   client.config.events.on(Event.SYNC_SYNCHRONIZED, (height) => {
-    logger.info(`Synchronized blockchain at height ${height.toNumber}`)
+    logger.info(`Synchronized blockchain at height ${height}`)
```

Comment:
Some side note: I would argue that (minimally) all log messages which are not initialization (in the sense of CLI setup and parsing) specific but rather react to some control flow events further down the line (so: e.g. the two events from above) are misplaced both here and in `bin/cli.ts` and just produce code redundancy and should rather be moved in to the core code base under `lib`.

Does that make sense or am I missing some aspect?

Reply:
yes that's a good idea :) will add in a commit